### ISSUE TITLE
python3-dev required?

### ISSUE THREAD

**Issue by sloretz** [ros2/python_cmake_module#3]
While building this on bionic I saw an error building `launch_testing_ament_cmake`

```
--- stderr: launch_testing_ament_cmake                                                                                                                                                            
CMake Error at /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find PythonLibs (missing: PYTHON_LIBRARIES PYTHON_INCLUDE_DIRS)
  (Required is at least version "3.5")
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.10/Modules/FindPythonLibs.cmake:262 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  /ros2_rolling/install/python_cmake_module/share/python_cmake_module/cmake/Modules/FindPythonExtra.cmake:133 (find_package)
  CMakeLists.txt:29 (find_package)


---
Failed   <<< launch_testing_ament_cmake [4.30s, exited with code 1]
```

It [depends on python_cmake_module](https://github.com/ros2/launch/blob/823ececc129c096c44e8883ea76cdfe49cb1841c/launch_testing_ament_cmake/package.xml#L19), but this repo `REQUIRE`s PythonLibs to be found

https://github.com/ros2/python_cmake_module/blob/80f6e549bba26f8c1f2b29149e5e158d8bef58e3/cmake/Modules/FindPythonExtra.cmake#L133

Maybe this package should depend on `python3-dev`? Or maybe `launch_testing_ament_cmake` should depend on `python3-dev`?

**Comment by clalancette** [comment]
There are actually two parts to this:
1.  What provides the FindPythonLibs.cmake file
1.  What provides the python libraries to be found

For the first one, it looks like that on Ubuntu, the package `cmake-data` provides `FindPythonLibs.cmake`.  I think we typically assume that that is installed, so I think we can ignore that one here.

For the second one, it looks like the actual package that installs Python.h and `/usr/lib/x86_64-linux-gnu/libpython3.8.so` (at least on Ubuntu 20.04) is the libpython3.8-dev package.  This in turn is depended on by python3.8-dev, which is depended on by python3-dev.  This same pattern holds for Ubuntu 18.04 as well.  On Fedora, the situation is a bit simpler in that `python3-devel` just contains the header files needed.

So that's the long way of saying that I agree with you; any place that calls `find_package(PythonLibs)` should depend on the python3-dev rosdep key.  And since this is one of those places, I think this package should be the one to depend on it.

**Comment by mikaelarguedas** [comment]
Shouldnt that be covered by https://github.com/ros2/python_cmake_module/blob/80f6e549bba26f8c1f2b29149e5e158d8bef58e3/package.xml#L17 ?

If I had to guess the issue comes from the fact that you used rosdep to install your dependencies and that this fix https://github.com/ros-infrastructure/rosdep/pull/753 still hasnt been released.

**Comment by sloretz** [comment]
> If I had to guess the issue comes from the fact that you used rosdep to install your dependencies and that this fix ros-infrastructure/rosdep#753 still hasnt been released.

That's 100% what happened. Thanks for pointing that out! Looks like there's nothing to fix in this repo so I'll close this.
